src/utils/timeline.rkt

```diff
@@ -60,6 +60,8 @@
       [(eq? *timeline-active-key* key)
        (set! *timeline-active-value* (cons val *timeline-active-value*))]
       [(not *timeline-active-key*)
+       (unless (pair? (unbox (*timeline*)))
+         (error 'timeline "Cannot push ~a to an empty timeline" *timeline-active-key*))
```

Comment:
I think you want key here, not timeline-active-key, which is always #f in this branch